django_fido/static/django_fido/js/u2f-registration.js

```diff
@@ -1,70 +1,71 @@
+    2: 'U2F request cannot be processed.',
+    3: 'Your configuration for U2F is not supported.',
+    4: 'The presented device is not eligible for this request.',
+    5: 'U2F request timed out.'
+}
 
 function addU2fError(message) {
-    var error_list = document.getElementById(U2F_ERROR_LIST_ID);
-    var new_item = document.createElement("li");
-    new_item.appendChild(document.createTextNode(message));
+    const error_list = document.getElementById(U2F_ERROR_LIST_ID);
+    const new_item = document.createElement("li");
+    new_item.classList.add('error');
+    new_item.appendChild(document.createTextNode(gettext(message)));
     error_list.appendChild(new_item);
 }
 
 function isU2fAvailabile() {
-    return !(typeof window.u2f === 'undefined');
-};
+    return (typeof window.u2f !== 'undefined');
+}
 
 function u2fResponseCallback(u2f_response) {
     if (u2f_response.errorCode) {
-        addU2fError('An error ' + u2f_response.errorCode + ' occured.');
+        addU2fError(U2F_ERROR_CODES[u2f_response.errorCode]);
         return
     }
-    var form = document.getElementById(U2F_FORM_ID);
+    const form = document.getElementById(U2F_FORM_ID);
     form.u2f_response.value = JSON.stringify(u2f_response);
     form.submit();
 }
 
-function u2fRegistrationRequestCallback() {
-    if (this.readyState == 4 && this.status == 200) {
-        var u2f_request = JSON.parse(this.responseText);
-        u2f.register(u2f_request.appId, u2f_request.registerRequests, u2f_request.registeredKeys, u2fResponseCallback, U2F_TIMEOUT);
+function processU2fRequest(formDataset) {
+    const http_request = new XMLHttpRequest();
+    http_request.onreadystatechange = () => {
+        if (http_request.readyState === 4 && http_request.status === 200) {
+            const u2f_request = JSON.parse(http_request.responseText);
+            if (formDataset.mode === U2F_AUTHENTICATION_REQUEST) {
+                u2f.sign(u2f_request.appId, u2f_request.challenge, u2f_request.registeredKeys, u2fResponseCallback, U2F_TIMEOUT);
+            } else if (formDataset.mode === U2F_REGISTRATION_REQUEST) {
+                u2f.register(u2f_request.appId, u2f_request.registerRequests, u2f_request.registeredKeys, u2fResponseCallback, U2F_TIMEOUT);
+            } else {
+                addU2fError("Unknown U2F request.");
+                return
+            }
+        }
     };
-}
-
-function u2fAuthenticationRequestCallback() {
-    if (this.readyState == 4 && this.status == 200) {
-        var u2f_request = JSON.parse(this.responseText);
-        u2f.sign(u2f_request.appId, u2f_request.challenge, u2f_request.registeredKeys, u2fResponseCallback, U2F_TIMEOUT);
-    };
-}
-
-function processU2fRequest(url, callback) {
-    var http_request = new XMLHttpRequest();
-    http_request.onreadystatechange = callback;
-    http_request.open('GET', url, true);
+    http_request.open('GET', formDataset.url, true);
     http_request.send();
```

Comment:
This can be probably simplified using fetch API: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API